api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -300,11 +307,16 @@ private void streamToOutput(final boolean streamTimeoutReached) {
                         partition != null &&
                         partition.isLookingForDeadLetter() &&
                         partition.getFailedCommitsCount() >= getContext().getMaxEventSendCount()) {
+
                     final ConsumedEvent failedEvent = toSend.remove(0);
-                    // todo: skip or to dlq
+
                     LOG.warn("Skipping event {} from partition {} due to failed commits count {}",
                             failedEvent.getPosition(), etp, partition.getFailedCommitsCount());
 
+                    if (getContext().getUnprocessableEventPolicy() == UnprocessableEventPolicy.DEAD_LETTER_QUEUE) {
+                        sendToDeadLetterQueue(failedEvent, partition.getFailedCommitsCount());
+                    }
```

Comment:
This (and `sendToDeadLetterQueue(...)`) mostly interact with context's fields.
What about instead delegate this implementation to the context?

I.e., smt. like (here):
```
getContext().processFailedEvent(failedEvent, partition.getFailedCommitsCount());
```

And the rest of the code would be implementation details in the `StreamingContext`.

Reply:
Yeah.  Or we may have something like a dedicated service for publishing to DLQ.  That's up to some refactoring.